python/catkin/builder.py

```diff
@@ -610,6 +611,11 @@ def build_cmake_package(
             '-DCMAKE_INSTALL_PREFIX=' + install_target
         ]
         cmake_cmd.extend(cmake_args)
+        cmake_cmd.append('-DCATKIN_VERSION:STRING=0.7.18')
+        if 'ROS_VERSION' in environ:
+            cmake_cmd.append('-DCATKIN_ROS_VERSION:STRING=' + environ['ROS_VERSION'])
+        if 'ROS_DISTRO' in environ:
+            cmake_cmd.append('-DCATKIN_ROS_DISTRO:STRING=' + environ['ROS_DISTRO'])
```

Comment:
It looks like `ROS_DISTRO` and `ROS_VERSION` can be gotten with `$ENV{ROS_DISTRO}` or `$ENV{ROS_VERSION}` https://cmake.org/cmake/help/v3.5/variable/ENV.html

Reply:
This is possible, however these extra CMake variables will help deal with potentially ambiguous versioning. For packages attempting to target different ROS versions and pure cmake builds it is better to be explicit.

Reply:
What causes the version to be ambiguous? Are you saying there is a case with this PR where `CATKIN_ROS_DISTRO` would be different from `$ENV{ROS_DISTRO}`?

Reply:
The build can either be done using Colcon, Catkin, or plain CMake, and they all have $ENV{ROS_DISTRO} set. There needs to be a way to detect that this is a ROS build and what tool is being used.

Reply:
Is this need based on an assumption that `colcon` means ROS 2, `catkin` means ROS 1, and `cmake` means no ROS at all? If so, I don't think that's a safe assumption to make.

Reply:
No, colcon can be used to build ROS 1 packages as well...

Reply:
> No, colcon can be used to build ROS 1 packages as well...

I'm trying to figure out what the need is. What does the cmake package want to do differently for each build tool?

> There needs to be a way to detect that this is a ROS build and what tool is being used.

It seems like `CATKIN_VERSION` and `COLCON_VERSION` are enough to say which tool called `cmake`, and `$ENV{ROS_DISTRO}` or `$ENV{ROS_VERSION}` are enough to say which version of ROS is being used. What are `CATKIN_ROS_DISTRO` and `COLCON_ROS_DISTRO` for?

Reply:
It is because Colcon and Catkin can be used outside of ROS! And yes, Colcon can be used to build ROS 1 packages. For a library maintainer there needs to be a way to EXPLICITLY detect what command has been executed. Right now the ROS-I team is doing acrobatics to attempt to detect and configure the different possible build conditions. It should not be this complicated.

Reply:
> For a library maintainer there needs to be a way to EXPLICITLY detect what command has been executed.

Why?

The package shouldn't care *how* `cmake` was involved. It should consider in what environment it is being built - e.g. has ROS been sourced or not.